doc/research/05_acting.md

```diff
@@ -0,0 +1,199 @@
+
+![Dangling carrot geometry](../assets/danglingcarrotgeometry.png)
+*source: [2]*
+
+$\frac{l_d}{sin(\alpha)}= 2R$, where $\alpha$ is the current heading error. Combining the two equations leads to our desired steering angle.
+
+$$
+\delta(t) = arctan(2L * \frac{sin(\alpha)}{l_d})
+$$
+
+This method can be tuned to work perfect for a certain speed, however we would like it to work in all speed regimes. We therefore calculate $l_d$ based on our current speed $v(t)$ and a tunable constant $K_d$ (, $l_d(t) = v(t) * K_d$). This leads to the final equation for our steering angle.
+
+$$
+\delta(t) = arctan(2L* \frac{sin(\alpha)}{K_d * v})
+$$
+
+- simple controller
+- ignores dynamic forces
+- assumes no-slip condition
+- possible improvement: vary the look-ahead distance based on vehicle velocity
+- not really suited for straights
+
+### Stanley
+
+The Stanley controller, named after an autonomous offroad race car, takes the front axle as a reference, while still using the bicycle model. In addition to looking at the heading error $\psi$, close to what pure pursuit does, stanley also looks at the cross track error $e_e$. The cross track error $e_e$ is defined as the distance between the reference point and the closest point on our trajectory.
+
+![Stanley error with heading and cross track error](../assets/stanleyerror.png)
+*source: [2]*
+
+The first part of our steering angle tries to correct for this error $arctan(\frac{k_e * e_e}{k_v*v})$ while the second part just corrects for our heading error $\psi$.
+
+This leads to the following equation for our steering angle.
+
+$$
+\delta(t) = \psi(t) + arctan(\frac{k_e*e_e(t)}{k_v*v(t)})
+$$
+
+With $k_e$ and $k_v$ being tuenable parameters for cross tracking error and speed respectively.
+
+### [MPC](https://en.wikipedia.org/wiki/Model_predictive_control) (Model Predictive Control) / receding horizon control
+
+The basic idea of MPC is to model the future behavior of the vehicle and compute an optimal control input that, minimizes an a priori defined cost functional.
+
+![MPC Controller](../assets/mpc.webp)
+*source: [5]*
+
+- cost function can be designed to account for driving comfort
+
+### [SMC](https://en.wikipedia.org/wiki/Sliding_mode_control) (sliding mode control)
+
+SMC systems are designed to drive the system states onto a particular surface in the state space, named sliding surface. Once the sliding surface is reached, sliding mode control keeps the states on the close neighborhood of the sliding surface.
+Real implementations of sliding mode control approximate theoretical behavior with a high-frequency and generally non-deterministic switching control signal that causes the system to chatter.
+
+![chattering](../assets/chattering.gif)
+*source: [9]*
+
+- simple
+- robust
+- stabile
+- disadvantage: chattering
```

Comment:
was hat chattering für Auswirkungen? Sind diese Nebenwirkungen akzeptierbar oder muss vielleicht ein controller Modell für sliding gefunden werden, das nicht das pendeln anfängt?

Reply:
Fixed in 7976481287682d94d4313ba3c6bbd41acdfe699c

Reply:
Es gibt keine vollständige Lösung für das Chattering-Problem. Das Fahrzeug würde mit einer SMC-Regelung zickzack fahren. Somit ist SMC eher ungeeignet für unsere Anwendung. Der Vollständigkeit halber würde ich es allerdings in der Recherche erwähnt lassen.